site/cds_rdm/inspire_harvester/writer.py

```diff
@@ -0,0 +1,228 @@
+                return False
+
+        current_rdm_records_service.publish(system_identity, draft["id"])
+        return True
+
+    def _get_existing_records(self, inspire_id):
+        """Find records that have already been harvested from INSPIRE."""
+        # for now checking only by inspire id
+        return current_rdm_records_service.search(
+            system_identity,
+            params={
+                "q": f"metadata.identifiers.identifier:{inspire_id} AND metadata.identifiers.scheme:inspire"
+            },
+        )
+
+    def _create_files(self, files_to_add, draft_id, inspire_id):
+        """Add and commit files to a draft."""
+        max_retries = 3
+        service = current_rdm_records_service
+        for _, file in files_to_add.items():
+            attempt = 0
+            url = file.pop("inspire_url")
+            while attempt < max_retries:
+                try:
+                    attempt += 1
+                    response = requests.get(url, stream=True)
+                    if response.status_code == 200:
+                        service.draft_files.init_files(
+                            system_identity,
+                            draft_id,
+                            [file],
+                        )
+                        content = BytesIO(response.content)
+                        service.draft_files.set_file_content(
+                            system_identity,
+                            draft_id,
+                            file["key"],
+                            content,
+                        )
+                        result = service.draft_files.commit_file(
+                            system_identity, draft_id, file["key"]
+                        )
+
+                        inspire_checksum = f"md5:{file['checksum']}"
+                        new_checksum = result.to_dict()["checksum"]
+                        assert inspire_checksum == new_checksum
+                        break
+                    else:
+                        current_app.logger.error(
+                            f"Retrieving file request failed on attempt {attempt}. Max "
+                            f"retries: {max_retries}. Status: {response.status_code}. "
+                            f"Response: {response.text}. URL: {url}. Filename: "
+                            f"{file['key']}. INSPIRE record id: {inspire_id}."
+                        )
+
+                        if attempt < max_retries:
+                            current_app.logger.error("Retrying in 1 minute...")
+                            time.sleep(60)
+                except Exception as e:
+                    current_app.logger.error(
+                        f"Retrieving file request failed on attempt {attempt}. Max retries: {max_retries}. Error {e}. URL: {url}. Filename: {file['key']}. INSPIRE record id: {inspire_id}."
+                    )
+
+                    try:
+                        if attempt < max_retries:
```

Comment:
not sure I understand why we are retrying both here and in line 210, could you explain?

Reply:
I have the same doubt 😅